### PR TITLE
Remove explicit snap deb dependencies

### PR DESCRIPTION
**Pull request by dragomirp**
# Issue
Since we are now installing the Patroni deb package properly from the PPA, we can remove some of the dependencies of the Snap

# Solution

# Context
* Unfortunately the space savings are marginal (2MB out of about 90MB)
* Updated with COS changes

# Testing
* Tested patronictl manually
* Draft PR canonical/postgresql-operator#132 to verify that integration tests still work

# Release Notes
Remove explicit deb dependencies

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -152,32 +152,19 @@ parts:
...
       - python3-pysyncobj
       - python3-boto3
-      - postgresql-common
-      - postgresql-client-14
+      - patroni
+      # Landscape deps
       - postgresql-contrib
       - postgresql-14-debversion
       - postgresql-plpython3-14
       - python-apt-dev
```
Comment (dragomirp) [inline snap/snapcraft.yaml:176]: We seem to need those for integration with Landscape. Split them away, so we know why they are there.

**Review by marceloneppel — APPROVED** [review]
LGTM!

File: snap/snapcraft.yaml (hunk 1 above)
Comment (taurus-forever) [reply snap/snapcraft.yaml:176]: JFYI, how Landscape uses python-apt in our charm? Do you mean extensions installation?

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -161,32 +161,19 @@ parts:
   postgres-debs:
     plugin: nil
     stage-packages:
-      - perl
-      - perl-base
```
Comment (taurus-forever) [inline snap/snapcraft.yaml:165]: Unfortunately, usrmerge, git, gpg-agent, init  depends on perl too:

```
snapcraft-charmed-postgresql-on-amd64-for-amd64-2584310 ../project# aptitude why perl-base
i   usrmerge Depends perl-base (>= 5.32.1-3)
...
i   git  Depends perl                         
i A perl Depends perl-base (= 5.34.0-3ubuntu1)
...
i   gpg-agent           Depends init-system-helpers (>= 1.52)
i A init-system-helpers Depends perl-base (>= 5.20.1-3)  
...
i   init                Depends init-system-helpers (>= 1.25)
i A init-system-helpers Depends perl-base (>= 5.20.1-3)
```

So, perl-base cannot be easily removed from SNAP...


BTW, why both core20 and core22 are shipped inside the snap?
```
sudo snapcraft build --shell-after
apt update && apt install ncdu
ncdu /

--- /snap ------
  232.5 MiB [##########] /core22                                                                                                                                                                                                                       
  198.3 MiB [########  ] /core20
  ... 
```

BTW2, is snapcraft shipped inside the snap?

```
sudo snapcraft build --shell-after

> snapcraft --version
snapcraft 7.3.1.post80+git89fd9ecb           
```

File: snap/snapcraft.yaml (hunk 1 above)
Comment (dragomirp) [reply snap/snapcraft.yaml:176]: I'm not sure of the specifics, but those used to be installed via `extra-packages` config options before we switched to the snap and the landscape fail without them installed.

File: snap/snapcraft.yaml (hunk 2 above)
Comment (dragomirp) [reply snap/snapcraft.yaml:165]: > So, perl-base cannot be easily removed from SNAP...

It should still be pulled as a dependency of other packages and should still be available in `/snap/charmed-postgresql/current/usr/bin/`

> BTW, why both core20 and core22 are shipped inside the snap?

I don't think the dev shell is fully isolated. We should be depending only on core22, but my understanding of snap esoterics is limited.

**Review by taurus-forever — APPROVED** [review]
After a long discussion, we can merge this, but we need to recheck what Landscape is doing inside our charm: https://www.postgresql.org/docs/current/plpython.html

**Pull request merged**